src/sage/graphs/graph.py

```diff
@@ -4476,6 +4476,82 @@ def is_factor_critical(self, matching=None, algorithm='Edmonds', solver=None, ve
         # The graph is factor critical if all vertices are marked even
         return len(even) == self.order()
 
+    def is_valid_hom(G, H, hom):
+    r"""
```

Comment:
Please follow the coding style, in particular for indentation

Reply:
For some reason copy-and-paste didn't work well.